schema/any.go

```diff
@@ -67,10 +67,13 @@ func (a *AnySchema) ValidateCompatibility(typeOrData any) error {
 	case reflect.Map:
 		return nil
 	default:
+		// Schema is not a primitive, slice, or map type, so check the complex types
+		// Explicitly allow object schemas since their reflected type can be a struct if they are struct mapped.
 		switch typeOrData.(type) {
-		case *AnySchema, *OneOfSchema[int64], *OneOfSchema[string]:
+		case *AnySchema, *OneOfSchema[int64], *OneOfSchema[string], *ObjectSchema:
```

Comment:
I assume there's already a test for `AnySchema`; but you added `ObjectSchema` ... was that covered?